src/core_plugins/metrics/public/components/custom_color_picker.js

```diff
@@ -0,0 +1,199 @@
+import React, { Component, PropTypes } from 'react';
+import reactCSS from 'reactcss';
+
+import { ColorWrap as colorWrap, Saturation, Hue, Alpha, Checkboard } from 'react-color/lib/components/common';
+import ChromeFields from 'react-color/lib/components/chrome/ChromeFields';
+import ChromePointer from 'react-color/lib/components/chrome/ChromePointer';
+import ChromePointerCircle from 'react-color/lib/components/chrome/ChromePointerCircle';
+import CompactColor from 'react-color/lib/components/compact/CompactColor';
+import color from 'react-color/lib/helpers/color';
+import shallowCompare from 'react-addons-shallow-compare';
+
+export class CustomColorPicker extends Component {
+  constructor(props) {
+    super(props);
+    this.shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1]);
```

Comment:
Just use `PureComponent` instead? See https://facebook.github.io/react/docs/react-api.html#react.purecomponent

And if not, this is difficult to read, so I'd rather implement it as we normally define lifecycle events (e.g. if you're not used to the lifecycle events it's difficult to remember that `arguments[0]` is props etc). Unless purecomponent, I'd do:

```
shouldComponentUpdate(nextProps, nextState) {
  return shallowCompare(nextProps, nextState)
}
```